rcl/test/rcl/test_client.cpp

```diff
@@ -277,3 +278,25 @@ TEST_F(TestClientFixture, test_client_bad_arguments) {
       &client, &client_request, &sequence_number)) << rcl_get_error_string().str;
   EXPECT_EQ(24, sequence_number);
 }
+
+TEST_F(TestClientFixture, test_client_init_fini_maybe_fail)
+{
+  const rosidl_service_type_support_t * ts = ROSIDL_GET_SRV_TYPE_SUPPORT(
+    test_msgs, srv, BasicTypes);
+  constexpr char topic_name[] = "chatter";
+  rcl_client_options_t default_client_options = rcl_client_get_default_options();
+
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    rcl_client_t client = rcl_get_zero_initialized_client();
+    rcl_ret_t ret = rcl_client_init(
+      &client, this->node_ptr, ts, topic_name, &default_client_options);
+    if (RCL_RET_OK == ret) {
```

Comment:
@brawner hmm, what if no error is injected within `rcl_client_init` but one makes it into `rcl_client_fini`?

Reply:
I added a second `rcl_client_fini` with an appropriate check for this case.

Reply:
I will point out that coping with incidental fault injection this way can get cumbersome fast. That's why at some point I suggested disabling fault injection entirely for the duration of these post-call checks. 

I won't block on this though.